src/wms/src/test/java/org/geoserver/wms/wms_1_1_1/CapabilitiesTest.java

```diff
@@ -795,6 +796,223 @@ public void testNonAdvertisedNonQueriableWithinGroup() throws Exception {
+        LayerGroupInfo group = null;
+        try {
+            group = createLakesPlacesLayerGroup(catalog, LayerGroupInfo.Mode.NAMED, null);
+            Document dom =
+                    dom(get("lakes_and_places/wms?request=GetCapabilities&version=1.1.0"), true);
+            // print(dom);
+
+            assertXpathEvaluatesTo("", "/WMT_MS_Capabilities/Capability/Layer/Name", dom);
+            assertXpathEvaluatesTo(
+                    "lakes_and_places", "/WMT_MS_Capabilities/Capability/Layer/Layer/Name", dom);
+        } finally {
+            if (group != null) {
+                catalog.remove(group);
+            }
+            info.getMetadata().remove(WMS.NO_ROOT_LAYER_IN_CAPABILITIES_KEY);
+            setAdvertised(catalog, MockData.LAKES, true);
+            setQueryable(catalog, MockData.LAKES, true);
+            setAdvertised(catalog, MockData.NAMED_PLACES, true);
+            setQueryable(catalog, MockData.NAMED_PLACES, true);
+        }
+    }
+
+    @Test
+    public void testRootLayerRemovedGroupConfig() throws Exception {
+        // make layers non advertised
+        Catalog catalog = getCatalog();
+        setAdvertised(catalog, MockData.LAKES, false);
+        setQueryable(catalog, MockData.LAKES, false);
+        setAdvertised(catalog, MockData.NAMED_PLACES, false);
+        setQueryable(catalog, MockData.NAMED_PLACES, false);
+        LayerGroupInfo group = null;
+        try {
+            group = createLakesPlacesLayerGroup(catalog, LayerGroupInfo.Mode.NAMED, null);
+            group.getMetadata().put(PublishedInfo.ROOT_IN_CAPABILITIES, true);
+            Document dom =
+                    dom(get("lakes_and_places/wms?request=GetCapabilities&version=1.1.0"), true);
+            // print(dom);
+
+            assertXpathEvaluatesTo(
+                    "lakes_and_places", "/WMT_MS_Capabilities/Capability/Layer/Name", dom);
+        } finally {
+            if (group != null) {
+                catalog.remove(group);
+            }
+            setAdvertised(catalog, MockData.LAKES, true);
+            setQueryable(catalog, MockData.LAKES, true);
+            setAdvertised(catalog, MockData.NAMED_PLACES, true);
+            setQueryable(catalog, MockData.NAMED_PLACES, true);
+        }
+    }
+
+    @Test
+    public void testRootLayerRemovedLayerConfig() throws Exception {
+        // make layers non advertised
+        Catalog catalog = getCatalog();
+        setAdvertised(catalog, MockData.LAKES, false);
+        setQueryable(catalog, MockData.LAKES, false);
+        LayerInfo layer = catalog.getLayerByName(MockData.LAKES.getLocalPart());
+
+        try {
+            layer.getMetadata().put(PublishedInfo.ROOT_IN_CAPABILITIES, true);
+            catalog.save(layer);
+            Document dom = dom(get("cite/Lakes/wms?request=GetCapabilities&version=1.1.0"), true);
+            // print(dom);
+
+            assertXpathEvaluatesTo("Lakes", "/WMT_MS_Capabilities/Capability/Layer/Name", dom);
```

Comment:
But if the layer is not advertised, it should not show up in the capabilities, the document should be empty....

Reply:
@aaime About this one, I think there is a bug in WMS 1.1.1 capabilities, it looks like advertised flag is not considered at all (while it is in 1.3.0). I would eventually open a separate JIRA for this. In the meantime I am going to cleanup my tests a bit (advertised / queryable was a copy/paste side effect, not needed for the tests themselves).

Reply:
That is strange, the advertised flag is processed at the catalog level, not directly in the capabilities... but it may be that the two caps generators are accessing the catalog in different ways, causing different behavior.

Reply:
The 1.3.0 transformer has an explicit filter for enabled AND advertised, in the 1.1.1 only I can only see one for enabled, I will check if the advertised flag eventually works at a lower (catalog) level

Reply:
That should only be perforamnce related, the AdvertisedCatalog class should be filtering away layers that are not advertised regardless (and there might well be a bug in there)